projects/packages/forms/src/blocks/contact-form/attributes.js

```diff
@@ -44,8 +44,12 @@ export default {
 			sendToSalesforce: false,
 		},
 	},
-	connectMailPoet: {
-		type: 'boolean',
-		default: false,
+	mailpoet: {
+		type: 'object',
+		default: {
+			enabledForForm: false,
+			listId: null,
+			listName: null,
+		},
```

Comment:
I've created a single 'complex' attribute like our salesforceData attribute to hold relevant MailPoet status and data.